SQL.Rmd

```diff
@@ -0,0 +1,1190 @@
+
+```sql
+ALTER TABLE my_table
+RENAME some_column TO new_column_name;
+```
+
+##### Table
+
+You can also rename an entire table with the `RENAME TO` clause. 
+
+```sql
+ALTER TABLE my_table
+RENAME TO new_table_name;
+```
+
+</br>
+
+### DROP TABLE
+
+The most destructive of all commands, the `DROP TABLE` statement removes the entire table including all data and metadata. Compared to a `DELETE` statement, `DROP TABLE` will even remove the table schema from the database. However, any other tables that exist and have some dependency (like a foreign key) will need to be updated to remove these before the database will allow the command to work. 
+
+```sql
+DROP TABLE (IF EXISTS) my_table;
+```
+
+Let's delete `smra_test`
+
+```{sql drop-table, connection="conn", exercise=TRUE}
+
+```
+
+```{sql drop-table-solution}
+DROP TABLE IF EXISTS smra_test;
+```
+
+
+## SQL in R
+
+The course, up until now, has been purely focussed on SQL. Hopefully this is helpful as the next step is bringing all of that knowledge into R. In this section we're going to explore connecting and querying databases from within R. That means that all the code can be in one place, you can write functions to directly interface with databases, and make efficiencies by utilising databases for your data rather than exporting and moving around data dumps in other formats. 
+
+### Connect
+
+The specification for connecting to and interacting with databases is outlined in the [`DBI`]() package. However, different databases require different R packages that provide more specific implementations, this includes:
+
+* [`RMySQL`](https://github.com/r-dbi/RMySQL) - for connecting to MySQL and MariaDB
+* [`RPostgreSQL`](https://github.com/tomoakin/RPostgreSQL) - for connecting to PostgreSQL and Redshift
+* [`RSQLite`](https://github.com/r-dbi/RSQLite) - for connecting to SQLite
+* [`odbc`](https://github.com/r-dbi/odbc) (Open Database Connectivity) - for connecting to Microsoft SQL Server and many other commercial databases
+* [`RODBC`](https://github.com/cran/RODBC) - less efficient than odbc but in some cases required as a fall-back.
+
+We'll continue with the `odbc` package and utilise SMRA as an example. When you're set-up with SMRA access, IT will also set up a DSN (Data Source Name), and under this alias all the details for interfacing with the database are stored. This layer of abstraction is the first security defence and also saves us some time.
+
+```r
+library(odbc)
+
+connection <- dbConnect(
+  odbc(),
+  dsn = "SMRA",
+  uid = .rs.askForPassword("SMRA Username: "),
+  pwd = .rs.askForPassword("SMRA Password: ")
+)
+```
+
+*Utilising the `.rs.askForPassword` functions provide a popup asking for details which are masked on entry and then passed back. This means that we don't have plain text passwords in our code. We could also use `keyring`, `config`, or environment variables to achieve this.*
```

Comment:
Not sure about suggesting environment variables, I wonder if it'll lead to people using something like 
pw <- "password" in their code